contracts/javascore/xcall/src/main/java/foundation/icon/btp/xcall/CallServiceImpl.java

```diff
@@ -0,0 +1,462 @@
+/*
+ * Copyright 2022 ICON Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foundation.icon.btp.xcall;
+
+import foundation.icon.btp.xcall.data.CSMessage;
+import foundation.icon.btp.xcall.data.CSMessageRequest;
+import foundation.icon.btp.xcall.data.CSMessageResponse;
+import foundation.icon.btp.xcall.data.CallRequest;
+import ibc.icon.interfaces.ICallServiceReceiver;
+
+import ibc.icon.interfaces.ICallServiceReceiverScoreInterface;
+import icon.proto.core.channel.Channel.Counterparty;
+import icon.proto.core.channel.Packet;
+import icon.proto.core.client.Height;
+import java.math.BigInteger;
+import score.Address;
+import score.Context;
+import score.RevertedException;
+import score.UserRevertException;
+import score.UserRevertedException;
+import score.annotation.EventLog;
+import score.annotation.External;
+import score.annotation.Optional;
+import score.annotation.Payable;
+
+
+public class CallServiceImpl extends AbstractCallService {
+
+    public CallServiceImpl(Address _ibc, BigInteger _timeoutHeight) {
+        this.ibcHandler.set(_ibc);
+        admin.set(Context.getOwner());
+        this.timeoutHeight.set(_timeoutHeight);
+    }
+
+    @External
+    public void setTimeoutHeight(BigInteger _timeoutHeight) {
+        onlyAdmin();
+        this.timeoutHeight.set(_timeoutHeight);
+    }
+
+    @External(readonly = true)
+    public BigInteger getTimeoutHeight() {
+        return this.timeoutHeight.getOrDefault(BigInteger.ZERO);
+    }
+
+    private void checkCallerOrThrow(Address caller, String errMsg) {
+        Context.require(Context.getCaller().equals(caller), errMsg);
+    }
+
+    private void onlyOwner() {
+        checkCallerOrThrow(Context.getOwner(), "OnlyOwner");
+    }
+
+    private void onlyAdmin() {
+        checkCallerOrThrow(this.admin(), "Only admin is allowed to call method");
+    }
+
+    private void onlyIBCHandler() {
+        checkCallerOrThrow(ibcHandler.get(), "Only IBCHandler allowed");
+    }
+
+//    private void checkService(String _svc) {
+//        Context.require(NAME.equals(_svc), "InvalidServiceName");
+//    }
```

Comment:
This can be remove for now, we instead check port and channel id

Reply:
removed